modules/core/src/smithy4s/http/internals/package.scala

```diff
@@ -56,19 +56,43 @@ package object internals {
       str: String
   ): Option[Vector[PathSegment]] = {
     str
-      .split('/')
-      .toVector
-      .filterNot(_.isEmpty())
-      .traverse(fromToString(_))
+      .split('?')
+      .headOption
```

Comment:
nitpick : here it's safe to call `.head` instead of `.headOption`, that'll save us an allocation